ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java

```diff
@@ -0,0 +1,756 @@
+
+                    @Override
+                    public Executor getRequestExecutor() {
+                        return executorService == null ? exchange.getIoThread().getWorker() : executorService;
+                    }
+
+                    @Override
+                    public String getProtocol() {
+                        return exchange.getProtocol().toString();
+                    }
+
+                    @Override
+                    public boolean isBlockingCaller() {
+                        return false;
+                    }
+
+                    @Override
+                    public EJBIdentifier getEJBIdentifier() {
+                        return ejbIdentifier;
+                    }
+
+    //                @Override
+                    public SecurityIdentity getSecurityIdentity() {
+                        return exchange.getAttachment(ElytronIdentityHandler.IDENTITY_KEY);
+                    }
+
+                    @Override
+                    public void writeException(@NotNull Exception exception) {
+                        if(identifier != null) {
+                            cancellationFlags.remove(identifier);
+                        }
+                        HttpServerHelper.sendException(exchange, httpServiceConfig, StatusCodes.INTERNAL_SERVER_ERROR, exception);
+                    }
+
+                    @Override
+                    public void writeNoSuchEJB() {
+                        if(identifier != null) {
+                            cancellationFlags.remove(identifier);
+                        }
+                        HttpServerHelper.sendException(exchange, httpServiceConfig, StatusCodes.NOT_FOUND, new NoSuchEJBException());
+                    }
+
+                    @Override
+                    public void writeCancelResponse() {
+                        if(identifier != null) {
+                            cancellationFlags.remove(identifier);
+                        }
+                        //we don't actually need to implement this method
+                    }
+
+                    @Override
+                    public void writeNotStateful() {
+                        if(identifier != null) {
+                            cancellationFlags.remove(identifier);
+                        }
+                        HttpServerHelper.sendException(exchange, httpServiceConfig, StatusCodes.INTERNAL_SERVER_ERROR, EjbHttpClientMessages.MESSAGES.notStateful());
+                    }
+
+                    @Override
+                    public void convertToStateful(@NotNull SessionID sessionId) throws IllegalArgumentException, IllegalStateException {
+                        throw new RuntimeException("nyi");
+                    }
+                });
+                if(handle != null && identifier != null) {
+                    cancellationFlags.put(identifier, handle);
+                }
+            });
+        }
+
+        private static String handleDash(String s) {
```

Comment:
Move handleDash to RemoteHTTPHandler as a common method.

Reply:
Done.